### DISCUSSION TITLE
Brew cleanup failing on openssl@1.1 (Catalina)

### DISCUSSION THREAD

**Discussion by nickbuker** [Homebrew/discussions#644] [Everyday usage] [answered]
Using MacOS Catalina and receiving this error when brew attempts run cleanup.

`Removing: /usr/local/Cellar/openssl@1.1/1.1.1g... (7,952 files, 12MB)
Error: Directory not empty @ dir_s_rmdir - /usr/local/Cellar/openssl@1.1/1.1.1g`

Please LMK what further information would be helpful.

**Comment by carlocab** [comment, marked as answer]
What is the output of `brew config`, `brew doctor`, and `brew info openssl@1.1`?

**Reply by nickbuker** [reply]
**brew config**
HOMEBREW_VERSION: 2.7.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: cd191a7f6c4c8e053336d0ea34a9886b12d06470
Last commit: 30 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: bedb5996fcad41697f6665c955b57258ed05b9e8
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.31.1
Xcode: N/A

**brew doctor**
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/more/.


Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/bukern/.pyenv/shims/python3.9-config
  /Users/bukern/.pyenv/shims/python-config
  /Users/bukern/.pyenv/shims/python3-config
  /Users/bukern/.pyenv/shims/python3.8-config

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/share/zsh/site-functions/_brew_cask


**brew info openssl@1.1**
openssl@1.1: stable 1.1.1i (bottled) [keg-only]
Cryptography and SSL/TLS Toolkit
https://openssl.org/
/usr/local/Cellar/openssl@1.1/1.1.1g (7,952 files, 12MB)
  Built from source
/usr/local/Cellar/openssl@1.1/1.1.1h (8,081 files, 22.3MB)
  Poured from bottle on 2020-10-23 at 10:09:31
/usr/local/Cellar/openssl@1.1/1.1.1i (8,067 files, 18.5MB)
  Poured from bottle on 2021-01-29 at 11:18:28
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/openssl@1.1.rb
License: OpenSSL
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /usr/local/etc/openssl@1.1/certs

and run
  /usr/local/opt/openssl@1.1/bin/c_rehash

openssl@1.1 is keg-only, which means it was not symlinked into /usr/local,
because macOS provides LibreSSL.

If you need to have openssl@1.1 first in your PATH, run:
  echo 'export PATH="/usr/local/opt/openssl@1.1/bin:$PATH"' >> ~/.zshrc

For compilers to find openssl@1.1 you may need to set:
  export LDFLAGS="-L/usr/local/opt/openssl@1.1/lib"
  export CPPFLAGS="-I/usr/local/opt/openssl@1.1/include"

==> Analytics
install: 705,490 (30 days), 2,272,453 (90 days), 8,122,989 (365 days)
install-on-request: 112,456 (30 days), 360,363 (90 days), 1,171,883 (365 days)
build-error: 0 (30 days)

**Reply by carlocab** [reply]
You have multiple `openssl@1.1` installations for some reason. Do
```
brew uninstall --ignore-dependencies openssl@1.1
```
thrice (yes, three times) then
```
brew install openssl@1.1
```

**Reply by nickbuker** [reply]
I think I resolved the issue by uninstalling all installs of openssl@1.1 and then reinstalling openssl@1.1. Version i uninstalled fine, but I had to sudo rm -rf versions g and h.

Afterwards brew cleanup was gave the following error:
Removing: /usr/local/Cellar/python@3.8/3.8.6... (1,157 files, 17.3MB)
Error: Directory not empty @ dir_s_rmdir - /usr/local/Cellar/python@3.8/3.8.6

I resolved this by nuking /usr/local/Cellar/python@3.8/3.8.6 and reinstalling python@3.8 

Everything seems happy now.

**Reply by nickbuker** [reply]
LMK how to mark as answered and thanks for the help! :)

**Reply by carlocab** [reply]
Sure; glad it's fixed. There should be a button somewhere around my original comment that does it...
